src/main/java/teammates/ui/webapi/GetFeedbackSessionsAction.java

```diff
@@ -93,7 +105,19 @@ public JsonResult execute() {
             }
         } else {
             feedbackSessionAttributes = logic.getFeedbackSessionsForCourse(courseId);
-            if (entityType.equals(Const.EntityType.INSTRUCTOR)) {
+            boolean isPotentiallyInvalidCourseId = feedbackSessionAttributes.isEmpty();
+            if (entityType.equals(Const.EntityType.STUDENT) && !isPotentiallyInvalidCourseId) {
```

Comment:
`isPotentiallyInvalidCourseId` is a misnomer. A course not having any sessions does not make it invalid.
```suggestion
            if (entityType.equals(Const.EntityType.STUDENT) && !feedbackSessionAttributes.isEmpty()) {
```

Reply:
I am aware, which is why I used "potentially". But my concern was that just using `!feedbackSessionAttributes.isEmpty()` would make it look like I am only checking for emptiness itself (which I think would be unnecessary) rather than the invalidity of the `courseId`. But if you think it's better I just leave it as `!feedbackSessionAttributes.isEmpty()` without a variable for it then that's fine too then.

Reply:
Sounds weird to me, empty feedback list doesn't really tell me much about whether courseId is valid. It would make more sense for the logic to be if list is empty, avoid making more unnecessary calls.

Reply:
Sounds weird to me, empty feedback list doesn't really tell me much about whether courseId is valid. It would make more sense for the logic to be if list is empty, avoid making more unnecessary calls.